BunqSdk/Http/ApiClient.cs

```diff
@@ -282,18 +290,47 @@ private static void AssertResponseSuccess(HttpResponseMessage responseMessage)
             var responseCode = (int) responseMessage.StatusCode;
             var responseBody = responseMessage.Content.ReadAsStringAsync().Result;
 
-            throw CreateApiExceptionRequestUnsuccessful(responseCode, responseBody);
+            throw CreateApiExceptionRequestUnsuccessful(
+                responseCode,
+                responseBody,
+                GetResponsId(responseMessage.Headers)
+                );
+        }
+
+        private static string GetResponsId(HttpHeaders allHeader)
+        {
+            if (allHeader.Contains(HEADER_RESPONSE_ID_UPPER_CASE))
```

Comment:
Maybe a method could be extracted here to avoid the conditional expressions.

Reply:
I cant seem to picture how an extracted method will prevent these conditional lines 🤔. Could you please provide a snippet with what you mean ?

Reply:
@OGKevin something like: 

```csharp
 private static string GetResponseId(HttpHeaders allHeader, IEnumerable<string> allHeaderResponse)
        {
            foreach (var headerResponse in allHeaderResponse)
            {
                if (allHeader.Contains(headerResponse))
                {
                    return allHeader.GetValues(headerResponse).First();
                }
            }

            throw new BunqException(ERROR_COULD_NOT_DETERMINE_RESPONSE_ID_HEADER);
        }
```

Actually... the loop brings needed extra complexity, in this case. Maybe it is better to stick with the conditional statements.

Reply:
Yes I think it is indeed better if we stick to the conditional statements! 👍